### ISSUE TITLE
Can't access SampleAPI using ConsoleResourceOwnerFlowReference

### ISSUE THREAD

**Issue by xamlsharp** [IdentityServer/IdentityServer4.Samples#60]
Hello,

I'm  currently working through your samples and can't figure out how to access the SampleAPI by using the "Console Resource Owner Flow Reference" project. I can successfully acquire a reference token, but keep getting a 401 error when it comes to executing the "CallService" method (which http's out to the sample api). I suspect I might need to somehow exchange the reference token for an access token, but the logic in the "CallService" method doesn't attempt to do that.

I'm raising this in case it is an issue with this particular sample. If not, could you point me to an example that shows how to call an api when using a reference token flow?

**Comment by brockallen** [comment]
The reference token is automatically de-referenced by the `UseIdentityServerAuthentication`. Which sample are you looking at?

**Comment by xamlsharp** [comment]
Hi Dominick,

Thanks for the swift reply. I appreciate it.

I've been through all of your quickstarts and was able to get them all working too. However, the problem I had was when pulling back your "clients" samples and working through them. 

At any rate, I have been able to solve the problem. Here is what happened:

In order to get a reference token, I changed the "roclient.reference" client configuration to use an AccessTokenType of "AccessTokenType.Reference". What's more, because your "clients" solution does not have an identity server project built in, I borrowed the one in your "MVC and API" solution (now Docker, I think). Unfortunately, what I did not realise is that when changing the token type to "reference", I also needed to make sure that the client secret was the **same** as the scope secret on the api. Any other attempt to access the sample api resulted in a 401.

Hence, for anyone else out there struggling with the same issue regarding using explicit reference tokens in these samples, makes sure you have a client secret **and** a scope secret that match.
